src/main/scala/cromwell/engine/SymbolStore.scala

```diff
@@ -79,4 +81,22 @@ class SymbolStore(binding: WdlBinding, inputs: Map[FullyQualifiedName, WdlValue]
         store.add(oldEntry.copyWithUpdatedValue(value))
     }
   }
+  
+  def addOutputValue(scope: String, name: String, maybeValue: Option[WdlValue], wdlType: WdlType): Try[Unit] = Try {
+    val key = SymbolStoreKey(scope, name, iteration = None, input = false)
+    
+    val maybeOldEntry = store.find { _.key == key }
+    maybeOldEntry match {
+      case Some(preexisting) =>
+        throw new IllegalArgumentException(s"Found unexpected preexisting symbol store entry with key '$key'")
```

Comment:
Why not return Success/Failure directly instead of wrapping the whole func in Try?